File: test/fixtures/ts-app-stack/discount.test.js
import assert from 'node:assert/strict';
import { test } from 'node:test';
import { stackedPrice } from './discount.js';

test('the percentage applies before the flat bulk discount', () => {
  // 10% off $100 is $90, then $5 flat off bulk orders is $85. This is the
  // seeded bug: applying the flat amount first gives $85.50 instead.
  assert.equal(stackedPrice(10000, 10, 500), 8500);
});

test('a flat discount with no percentage is a plain subtraction', () => {
  assert.equal(stackedPrice(10000, 0, 500), 9500);
});

File: test/fixtures/go-app-stack/discount_test.go
package discount

import "testing"

func TestPercentageAppliesBeforeFlatBulkDiscount(t *testing.T) {
	// 10% off $100 is $90, then $5 flat off bulk orders is $85. This is the
	// seeded bug: applying the flat amount first gives $85.50 instead.
	if got := StackedPrice(10000, 10, 500); got != 8500 {
		t.Errorf("StackedPrice(10000, 10, 500) = %d, want 8500", got)
	}
}

func TestFlatDiscountWithNoPercentageIsPlainSubtraction(t *testing.T) {
	if got := StackedPrice(10000, 0, 500); got != 9500 {
		t.Errorf("StackedPrice(10000, 0, 500) = %d, want 9500", got)
	}
}
